### DISCUSSION TITLE
`vercel dev` with Python runtime downloads `index.py` file instead of running function

### DISCUSSION THREAD

**Discussion by isaacharrisholt** [vercel/community#2768] [Bug Report]
### Page to Investigate

https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/python

### Steps to Reproduce

I have a monorepo containing a number of Vercel projects. One of said Vercel projects is for functions only. I'm trying to create a Python function within the `/api/test` directory, like so:

```
repo
└── functions (Vercel project root)
    └── api
        └── test
            └── index.py
```

The Python function is using FastAPI:

```python
from fastapi import FastAPI

app = FastAPI()


@app.get("/api/test")
def hello_world():
    return "Hello, world!"
```

However, I've had this issue with the HTTP example provided [in the docs](https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/python).

Essentially, when I run `vercel dev` from the root of my repo (which has been `vercel link`ed to my functions project) and navigate to `localhost:3000/api/test`, my browser tries to download the `index.py` file instead of running the function.

If I rename `index.py` to `foo.py` and head to `/api/test/foo`, I get a 502 error:

```
using Asynchronous Server Gateway Interface (ASGI)
<class 'TypeError'>
TypeError: Object prototype may only be an Object or null: undefined
    at Function.setPrototypeOf (<anonymous>)
    at new LambdaError (/home/isaac/.nvm/versions/node/v18.15.0/lib/node_modules/vercel/dist/index.js:8710:16)
    at Lambda.<anonymous> (/home/isaac/.nvm/versions/node/v18.15.0/lib/node_modules/vercel/dist/index.js:8843:27)
    at Generator.next (<anonymous>)
    at fulfilled (/home/isaac/.nvm/versions/node/v18.15.0/lib/node_modules/vercel/dist/index.js:8757:58)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
```

None of this happens when deployed though - it all works just fine.

Any suggestions as to what I'm doing wrong?

**Comment by hassanpasha5630** [comment]
I was having the same issue so I did something like this     "fastapi-dev": "pip3 install -r requirements.txt && python3 -m uvicorn api.index:app --reload",

I would just edit the run command for example if I updated index to foo I would do api.foo:app this worked for me

Hope this helps

**Comment by SaraswatAnushka** [comment]
i am not using fastapi but I am having the same issue
Can anybody help, I don't know how to work with flask and this deployment

**Comment by hammaddaftab** [comment]
If you set Vercel project root to default that is to the root of repo (just don't modify vercel root path) then you can go with:
{
  "builds": [
    {
      "src": "/functions/api/index.py",
      "use": "@vercel/python"
    }
  ],
  "rewrites": [
    {
      "source": "/(.*)",
      "destination": "/functions/api/index.py"
    }
  ]
}
